tesseract_kinematics/include/tesseract_kinematics/ikfast/ikfast_inv_kin.h

```diff
@@ -141,14 +143,25 @@ class IKFastInvKin : public InverseKinematics
             std::vector<std::string> link_names,
             std::vector<std::string> active_link_names,
             tesseract_common::KinematicLimits limits,
-            std::vector<Eigen::Index> redundancy_indices);
+            std::vector<Eigen::Index> redundancy_indices,
+            std::vector<std::vector<double>> free_joint_combos = {});
 
   /**
    * @brief Checks if kinematics has been initialized
    * @return True if init() has completed successfully
    */
   bool checkInitialized() const;
 
+  /**
+   * @brief Generates all possible combinations of joint states and stores it to the free_joint_combos_ class member
+   * Example: Given 2 free joints, wanting to sample the first joint at 0, 1, and 2 and the second joint at 3 and 4
+   * the input would be [[0, 1, 2][3,4]] and it would generate [[0,3][0,4][1,3][1,4][2,3][2,4]]
+   * @param free_joint_samples A vector of vectors in which the lower level vectors each represent all of a single
+   * joint's possible positions to be sampled
+   * @return
+   */
+  void generateAllPossibleCombinations(const std::vector<std::vector<double>> free_joint_samples);
```

Comment:
```suggestion
  void generateAllFreeJointStateCombinations(const std::vector<std::vector<double>> free_joint_samples);
```

Similarly pedantic, but adds more clarity. The documentation is helpful though

Reply:
Agreed, this matches the other renaming you requested